taskvine/src/manager/vine_file.c

```diff
@@ -37,6 +40,107 @@ In each of the other file types, a similar approach is taken,
 including a hash and a name where one is known, or another
 unique identifier where no name is available.
 */
+int do_command(char * command, char **buffer){
+	FILE * stream = popen(command, "r");
+	int result = 0;
+        if(stream){
+                copy_stream_to_buffer(stream, buffer, 0);
+                int exit_status = pclose(stream);
+                if(exit_status==0){
+                        result = 1;
+                }
+                else{
+                        // command failed
+                        result = 0;
+                }
+        }
+        else{
+		result = 0;
+                // couldn't execute command
+        }
+	return result;
+}
+
+const char *make_url_cached_name(const struct vine_file *f)
+{
+	int result = 0;
+
+	int STR_MAX = 256;
+	char line[STR_MAX];
+	char hash_src[STR_MAX];
+	char * buffer;
+	unsigned char digest[MD5_DIGEST_LENGTH];
+
+	char tmp[] = "curl_tmp.XXXXXX";
+	mkstemp(tmp); 
+
+	char * command = string_format("curl -I -sSL --stderr /dev/stdout -o \"%s\" \"%s\"", tmp, f->source);
+	result = do_command(command, &buffer);
```

Comment:
This seems overly complicated.  Just use `popen` to run the command and read its output with `fgets` and `sscanf`.  That way you don't have to generate a temporary file and clean it up.